apps/self-hosted/hosting/scripts/hivesigner-redirect-uris.py

```
@@ -131,6 +271,113 @@ def write_private(contents: str, path: str | None) -> str:
+
+    Deliberately sends nothing. The API decides what to enable by reading the
+    chain itself, so this cannot ask it to enable an instance that is not
+    registered, and neither can anything that gets hold of this script.
+    """
+    url = api_base.rstrip("/") + "/v1/internal/hivesigner/reconcile"
+    request = urllib.request.Request(
+        url,
+        data=b"",
+        method="POST",
+        headers={"Content-Type": "application/json", "x-internal-secret": secret},
+    )
```

Ruff (0.16.0)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)